src/codegen/type/decimal_type.cpp

```diff
@@ -141,7 +141,7 @@ struct CompareDecimal : public TypeSystem::SimpleComparisonHandleNull {
                            const Value &right) const override {
     // For integer comparisons, just subtract left from right and cast the
     // result to a 32-bit value
-    llvm::Value *diff = codegen->CreateSub(left.GetValue(), right.GetValue());
+    llvm::Value *diff = codegen->CreateFSub(left.GetValue(), right.GetValue());
```

Comment:
I wonder why the IR verification doesn't catch instructions which value types don't match.

Reply:
It will. The problem was that this code path was not being exercised in the tests.